muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/communicator/CommunicatorController.java

```diff
@@ -437,9 +479,17 @@ public int compare(CommunicatorMessage o1, CommunicatorMessage o2) {
     
     return new ArrayList<>(result);
   }
-
+  
   public CommunicatorMessageRecipient updateRead(CommunicatorMessageRecipient recipient, boolean value) {
-    return communicatorMessageRecipientDAO.updateRecipientRead(recipient, value);
+    try {
+      return communicatorMessageRecipientDAO.updateRecipientRead(recipient, value);
+    } finally {
+      UserEntity userEntity = userEntityController.findUserEntityById(recipient.getRecipient());
+      List<CommunicatorMessage> communicatorMessages = listMessagesByMessageId(userEntity, recipient.getCommunicatorMessage().getCommunicatorMessageId());
+      for(CommunicatorMessage message : communicatorMessages) {
```

Comment:
Probably enough to just index the single message from recipient.getCommunicatorMessage()